Sources/JWTKit/JWTKeyCollection.swift

```diff
@@ -205,8 +219,21 @@ public actor JWTKeyCollection: Sendable {
     {
         let header = try defaultJWTParser.parseHeader(token)
         let kid = header.kid.flatMap { JWKIdentifier(string: $0) }
-        let signer = try self.getSigner(for: kid, alg: header.alg)
-        return try await signer.verify(token)
+        var signer = try self.getSigner(for: kid, alg: header.alg)
+
+        do {
+            return try await signer.verify(token)
+        } catch {
+            if self.shouldIterateKeys {
+                for (kid, _) in self.storage {
+                    do {
+                        signer = try self.getSigner(for: kid, alg: header.alg)
```

Comment:
What happens if a signer doesn't have a `kid`?

Reply:
It's marked as default. Adding a different signer without a kid will override the one before